autogen/agentchat/conversable_agent.py

```diff
@@ -1781,12 +1995,80 @@ def register_hook(self, hookable_method: Callable, hook: Callable):
             hookable_method: A hookable method implemented by ConversableAgent.
             hook: A method implemented by a subclass of AgentCapability.
         """
-        assert hookable_method in self.hook_lists, f"{hookable_method} is not a hookable method."
+        if hookable_method not in self.hook_lists:
```

Comment:
@rickyloynd-microsoft Can you please check out these two changes? Asserts should not be used in code but only in tests as they can be removed by the compiler (https://codereviewdoctor.medium.com/you-might-be-using-assert-wrong-6a9daf821a03)

Reply:
Thanks for pointing this out. I think it would be better to replace the two asserts in this function by throwing ValueError exceptions, rather than failing silently.

Reply:
Thanx for the feedback :)